### ISSUE TITLE
role_name_ prefix conforming variables erroneously flagged in include_role task

### ISSUE THREAD

**Issue by mklassen** [ansible/ansible-lint#3491]
##### Summary

var-naming[no-role-prefix] flags variables being using in `include_role` since 6.16.1

```yaml
- name: Problem task
  ansible.builtin.include_role:
    name: org.fq.custom_role
  vars:
    custom_role_variable: something
```

`custom_role_variable` is flagged as not conforming to the role_name_ prefix rule. 

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.16.2 using ansible-core:2.14.6 ruamel-yaml:0.17.26 ruamel-yaml-clib:0.2.7
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Use role variables with to role_name_ prefix  in `ansible.builtin.include_role` task.

```console (paste below)
ansible-galaxy collection init my_namespace.my_collection
cd my_namespace/my_collection
echo -e "---\nrequires_ansible: '>=2.11.0'" > meta/runtime.yml
mkdir -p roles/my_role/tasks
cat <<EOT >> roles/my_role/tasks/main.yml
- name: Task
  ansible.builtin.include_role:
    name: my_namespace.my_collection.my_role2
  vars:
    my_role2_variable: something
EOT
ansible-lint
```

##### Desired Behavior

No var-naming[no-role-prefix] error is generated

```
WARNING  Listing 5 violation(s) that are fatal
galaxy[no-changelog]: No changelog found. Please add a changelog file. Refer to the galaxy.md file for more info.
galaxy.yml:1

galaxy[tags]: galaxy.yaml must have one of the required tags: ['application', 'cloud', 'database', 'infrastructure', 'linux', 'monitoring', 'networking', 'security', 'storage', 'tools', 'windows']
galaxy.yml:1

yaml[indentation]: Wrong indentation: expected at least 1
galaxy.yml:19

yaml[indentation]: Wrong indentation: expected at least 1
galaxy.yml:29

yaml[empty-lines]: Too many blank lines (1 > 0)
galaxy.yml:69

Read documentation for instructions on how to ignore specific rule violations.

                 Rule Violation Summary
 count tag                  profile rule associated tags
     1 yaml[empty-lines]    basic   formatting, yaml
     2 yaml[indentation]    basic   formatting, yaml
     1 galaxy[no-changelog] shared  metadata
     1 galaxy[tags]         shared  metadata

Failed after min profile: 5 failure(s), 0 warning(s) on 4 files.
```

##### Actual Behavior

var-naming[no-role-prefix] error is generated

```paste below
WARNING  Listing 6 violation(s) that are fatal
galaxy[no-changelog]: No changelog found. Please add a changelog file. Refer to the galaxy.md file for more info.
galaxy.yml:1

galaxy[tags]: galaxy.yaml must have one of the required tags: ['application', 'cloud', 'database', 'infrastructure', 'linux', 'monitoring', 'networking', 'security', 'storage', 'tools', 'windows']
galaxy.yml:1

yaml[indentation]: Wrong indentation: expected at least 1
galaxy.yml:19

yaml[indentation]: Wrong indentation: expected at least 1
galaxy.yml:29

yaml[empty-lines]: Too many blank lines (1 > 0)
galaxy.yml:69

var-naming[no-role-prefix]: Variables names from within roles should use role_name_ as a prefix. (vars: my_role2_variable)
roles/my_role/tasks/main.yml:5 Task/Handler: Task

Read documentation for instructions on how to ignore specific rule violations.

                    Rule Violation Summary
 count tag                        profile rule associated tags
     1 var-naming[no-role-prefix] basic   idiom
     1 yaml[empty-lines]          basic   formatting, yaml
     2 yaml[indentation]          basic   formatting, yaml
     1 galaxy[no-changelog]       shared  metadata
     1 galaxy[tags]               shared  metadata

Failed after min profile: 6 failure(s), 0 warning(s) on 6 files.
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by Rylon** [comment]
I think this is the same as I reported here https://github.com/ansible/ansible-lint/issues/3465 ?

It's been fixed, but just waiting for the next release to go out.
